android/app/src/main/kotlin/com/yubico/authenticator/oath/OathManager.kt

```diff
@@ -554,29 +555,50 @@ class OathManager(
             NULL
         }
 
-    private suspend fun requestRefresh() =
+    private suspend fun requestRefresh() {
+
+        val clearCodes = {
+            val currentCredentials = oathViewModel.credentials.value
+            oathViewModel.updateCredentials(currentCredentials?.associate {
+                it.credential to null
+            } ?: emptyMap())
+        }
+
         appViewModel.connectedYubiKey.value?.let { usbYubiKeyDevice ->
-            useOathSessionUsb(usbYubiKeyDevice) { session ->
-                try {
-                    oathViewModel.updateCredentials(calculateOathCodes(session))
-                } catch (apduException: ApduException) {
-                    if (apduException.sw == SW.SECURITY_CONDITION_NOT_SATISFIED) {
-                        logger.debug("Handled oath credential refresh on locked session.")
-                        oathViewModel.setSessionState(
-                            Session(
-                                session,
-                                keyManager.isRemembered(session.deviceId)
+            try {
+                useOathSessionUsb(usbYubiKeyDevice) { session ->
+                    try {
+                        oathViewModel.updateCredentials(calculateOathCodes(session))
+                    } catch (ioException: IOException) {
```

Comment:
Do we need this catch? Wouldn't removing it cause the same exception to instead be caught on line 592?

Reply:
Good catch - it is enough to catch in the outer scope. I updated the code now and tested on devices.